android/Staccato_AN/app/src/main/java/com/woowacourse/staccato/presentation/timeline/TimelineFragment.kt

```diff
@@ -2,18 +2,51 @@ package com.woowacourse.staccato.presentation.timeline
 
 import android.os.Bundle
 import android.view.View
+import androidx.lifecycle.ViewModelProvider
 import androidx.navigation.fragment.findNavController
 import com.woowacourse.staccato.R
 import com.woowacourse.staccato.databinding.FragmentTimelineBinding
 import com.woowacourse.staccato.presentation.base.BindingFragment
+import com.woowacourse.staccato.presentation.timeline.adapter.TimelineAdapter
+
+class TimelineFragment :
+    BindingFragment<FragmentTimelineBinding>(R.layout.fragment_timeline),
+    TimelineHandler {
+    private lateinit var viewModel: TimelineViewModel
```

Comment:
아래 처럼 KTX를 활용해보는 건 어떨까요?
KTX를 활용하면 setUpViewModel() 함수도 불필요해집니다!

```suggestion
    private var viewModel: TimelineViewModel by viewModels {
    TimelineViewModelFactory(TempTimelineRepository())
    }
```

<img width="537" alt="image" src="https://github.com/user-attachments/assets/d89b9682-142a-46b3-9364-ce4997e37229">
setUpViewModel() 함수가 불필요해지는 이유는 viewModels은 매개변수로 factoryProducer를 바로 넣어줄 수 있기 때문입니다!

<br>
<br>

Reply:
미션에서도 진행했던 내용이네요...! 😮 
`viewModels()` 메서드가 내부적으로 저렇게 동작하는 것은 오늘 처음 알았습니다.
라이브러리 코드를 보니 미션에서의 코드가 이해가 되네요~ 감사합니다! 😄 
추후에 꼭 리팩터링 해보겠습니다!!